src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,92 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> components = new HashMap<>();
```

Comment:
Let's check instances map before new instance creation. This ensures that you do not create a new instance if one already exists, which would be more efficient.